app/jobs/fetch_constituencies_job.rb

```diff
@@ -62,4 +64,18 @@ def regions
   def example_postcodes
     @example_postcodes ||= YAML.load_file(Rails.root.join("data", "example_postcodes.yml"))
   end
+
+  def parliaments
+    Parliament.all
+  end
+
+  def update_constituencies_parliaments(constituency)
+    parliaments.each do |parliament|
+      if constituency.intersects_parliament?(parliament)
+        ConstituenciesParliament.find_or_create_by!(constituency_id: constituency.id, parliament_id: parliament.id)
```

Comment:
This is why you're needing to specify the model - you're using as a model and the default class that Rails creates is something like `Parliament::Constituencies_HABTM`. If you do what I suggest in the other comment you'll be able to delete the model as all the work is done through the association.